packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -310,12 +318,42 @@ export function TransferPanel() {
       return
     }
 
+    // ETH transfers aren't enabled yet. Safety check, shouldn't be able to get here.
+    if (isSmartContractWallet && !selectedToken) {
+      console.error("ETH transfers aren't enabled for smart contract wallets.")
+      return
+    }
+
+    // SC wallet transfer requests are sent immediatelly, delay it to give user an impression of a tx sent
+    const showDelayedSCTxRequest = () =>
+      setTimeout(() => {
+        setTransferring(false)
+        setShowSCWalletTooltip(true)
+      }, 3000)
+
     const setTransferring = (payload: boolean) =>
       dispatch({ type: 'layout.set_is_transferring', payload })
 
     setTransferring(true)
 
     try {
+      const isDestinationAddressSmartContract = await addressIsSmartContract(
+        destinationAddress,
+        isDepositMode ? l2Provider : l1Provider
+      )
+
+      if (
+        // Invalid address
+        (destinationAddress && !isAddress(destinationAddress)) ||
+        // Destination address not matching the connected wallet type
+        (destinationAddress &&
+          isSmartContractWallet !== isDestinationAddressSmartContract)
```

Comment:
Doing an early return if there's no value for `destinationAddress` makes things a bit more readable:

```suggestion
      if (!destinationAddress) {
        throw new Error(
          `Couldn't initiate the transfer. Destination address not provided.`
        )
      }

      if (
        // Invalid address
        !isAddress(destinationAddress) ||
        // Destination address not matching the connected wallet type
        isSmartContractWallet !== isDestinationAddressSmartContract
      )
```

Reply:
The only issue with that is we want to allow an undefined destinationAddress for EOA, that's why I'm checking for validity not whether it's defined

Reply:
Update: refactored it to check for `destinationAddress` first